crates/sui-json-rpc-types/src/sui_object.rs

```diff
@@ -442,31 +470,31 @@ impl
 impl SuiObjectResponse {
     /// Returns a reference to the object if there is any, otherwise an Err if
     /// the object does not exist or is deleted.
-    pub fn object(&self) -> UserInputResult<&SuiObjectData> {
-        match &self {
-            Self::Deleted(oref) => Err(UserInputError::ObjectDeleted {
-                object_ref: oref.to_object_ref(),
-            }),
-            Self::NotExists(id) => Err(UserInputError::ObjectNotFound {
-                object_id: *id,
-                version: None,
-            }),
-            Self::Exists(o) => Ok(o),
+    pub fn object(&self) -> Result<&SuiObjectData, SuiObjectResponseError> {
+        let data = &self.data;
+        let error = self.error.clone();
+        if let Some(data) = data {
+            Ok(data)
+        } else if let Some(error) = error {
+            Err(error)
+        } else {
+            // We really shouldn't reach this code block since either data, or error field should always be filled.
+            Err(SuiObjectResponseError::Unknown)
```

Comment:
Can we use `error!()` to log the id here?

Reply:
No because it would've been in the data fields. This case only happens when SuiObjectResponse has None error and data fields.